witchcraft/wdebug/handlers.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2020 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wdebug
+
+import (
+	"context"
+	"io"
+	"runtime/pprof"
+	"time"
+
+	"github.com/palantir/conjure-go-runtime/v2/conjure-go-contract/codecs"
+	"github.com/palantir/conjure-go-runtime/v2/conjure-go-contract/errors"
+	"github.com/palantir/pkg/metrics"
+	werror "github.com/palantir/witchcraft-go-error"
+	wparams "github.com/palantir/witchcraft-go-params"
+)
+
+const (
+	DiagnosticTypeCPUProfile1MinuteV1 DiagnosticType = "go.profile.cpu.1minute.v1"
+	DiagnosticTypeHeapProfileV1       DiagnosticType = "go.profile.heap.v1"
+	DiagnosticTypeAllocsProfileV1     DiagnosticType = "go.profile.allocs.v1"
+	DiagnosticTypeGoroutinesV1        DiagnosticType = "go.goroutines.v1"
+	DiagnosticTypeMetricNamesV1       DiagnosticType = "metric.names.v1"
+)
+
+var diagnosticHandlers = map[DiagnosticType]DiagnosticHandler{
+	DiagnosticTypeCPUProfile1MinuteV1: handlerCPUProfile1MinuteV1{},
+	DiagnosticTypeHeapProfileV1:       handlerHeapProfileV1{},
+	DiagnosticTypeAllocsProfileV1:     handlerAllocsProfileV1{},
+	DiagnosticTypeGoroutinesV1:        handlerGoroutinesV1{},
+	DiagnosticTypeMetricNamesV1:       handlerMetricNamesV1{},
+}
+
+type DiagnosticHandler interface {
```

Comment:
Thoughts on keeping this private until we decide how to advertise everything in case we need to change anything and avoid breaks?
Alternatively we could move the entirety of wdebug to internal until the details are hashed out. I am not too concerned with the existing public api, so an internal move might not be worth it.

Reply:
Sure, moving to internal where we can always pull it back out